From 5a6437e44269534bfde0c35cc2c7bdef087b70e8 Mon Sep 17 00:00:00 2001
From: Dhruv Govil <dgovil2@apple.com>
Date: Sat, 23 Nov 2024 13:41:57 -0800
Subject: [PATCH] An upcoming change in mainline clang will be validating
 templated code even when the specific path is not taken during compilation.

This fix removes on such unused path from SdfChildrenProxy in the assignment operator where there is no implementation of _Set
---
 pxr/usd/sdf/childrenProxy.h | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/pxr/usd/sdf/childrenProxy.h b/pxr/usd/sdf/childrenProxy.h
index f51552853d..c28d8f03c2 100644
--- a/pxr/usd/sdf/childrenProxy.h
+++ b/pxr/usd/sdf/childrenProxy.h
@@ -52,13 +52,6 @@ class SdfChildrenProxy {
             return *_pos;
         }
 
-        template <class U>
-        _ValueProxy& operator=(const U& x)
-        {
-            _owner->_Set(*_pos, x);
-            return *this;
-        }
-
         bool operator==(const mapped_type& other) const
         {
             return *_pos == other;
